rajawali/src/main/java/c/org/rajawali3d/engine/EngineImpl.java

```diff
@@ -220,105 +244,104 @@ public void onRenderSurfaceCreated(EGLConfig config, GL10 gl, int width, int hei
 
         ++frameCount;
         if (frameCount % 50 == 0) {
             long now = System.nanoTime();
-            double elapsedS = (now - startTime) / 1.0e9;
+            double elapsedS = (now - fpsStartTime) / 1.0e9;
             double msPerFrame = (1000 * elapsedS / frameCount);
             lastMeasuredFPS = 1000 / msPerFrame;
 
             frameCount = 0;
-            startTime = now;
+            fpsStartTime = now;
 
             if (fpsUpdateListener != null)
                 fpsUpdateListener.onFPSUpdate(lastMeasuredFPS); // Update the FPS listener
         }
     }
 
-    @Override
-    public void onOffsetsChanged(float xOffset, float yOffset, float xOffsetStep, float yOffsetStep, int xPixelOffset, int yPixelOffset) {
-
-    }
+    protected void startFrame(double deltaTime) {
 
-    @Override
-    public void onTouchEvent(MotionEvent event) {
+        // TODO Any engine-level frame tasks, e.g. for adding/removing models or views?
 
+        try {
+            for (RenderModel renderModel : renderModels) {
+                renderModel.onFrameStart(deltaTime);
+            }
+            for (RenderView renderView : renderViews) {
+                renderView.onFrameStart(deltaTime);
+            }
+        } catch (InterruptedException e) {
+            Log.e(TAG, "startFrame(): Frame skipped due to thread interruption.");
+        }
     }
 
-    /**
-     * Called by {@link #onRenderFrame(GL10)} to render the next frame. This is called prior to the current scene's
-     * {@link Scene#render(long, double)} method.
-     *
-     * @param ellapsedRealtime {@code long} The total ellapsed rendering time in milliseconds.
-     * @param deltaTime        {@code double} The time passes since the last frame, in seconds.
-     */
-    protected void onRender(final long ellapsedRealtime, final double deltaTime) throws InterruptedException {
-        render(ellapsedRealtime, deltaTime);
+    protected void renderViews() {
+        try {
+            for (RenderView renderView : renderViews) {
+                renderView.onRenderView();
+            }
+        } catch (InterruptedException e) {
+            Log.e(TAG, "renderViews(): Frame skipped due to thread interruption.");
+        }
     }
 
-    /**
-     * Called by {@link #onRender(long, double)} to render the next frame.
-     *
-     * @param ellapsedRealtime {@code long} Render ellapsed time in milliseconds.
-     * @param deltaTime        {@code double} Time passed since last frame, in seconds.
-     */
-    protected void render(final long ellapsedRealtime, final double deltaTime) throws InterruptedException {
-        if (currentRenderable != null) {
-            currentRenderable.render(ellapsedRealtime, deltaTime);
-        } else {
-            Log.w(TAG, "No renderable has been set!");
+    protected void endFrame(double deltaTime) {
+        try {
+            // TODO not sure if order matters here, just swapping from startFrame() on principle...
```

Comment:
I imagine this is something that will become apparent as the implementations are fleshed out, but I'm inclined to suspect that the order is backwards. From my understanding, I am thinking we will want `RenderView`s to receive the callback prior to the `RenderModel`s. Funny side joke about that: [Flesh vs Flush](http://theoatmeal.com/comics/flesh_vs_flush)